### ISSUE TITLE
UI: Deletion of packages

### ISSUE THREAD

**Issue by LordMike** [loic-sharma/BaGet#137]
### Do you want to request a *feature* or report a *bug*?

Feature

### What did you expect to see?

* A delete button next to a package (all versions)
* A delete button next to a package version
* A delete button for many packages, perhaps all at once

**Comment by loic-sharma** [comment]
> In other words, are there any high level plans for the web ui?

Yup! I intend to recreate most of NuGet.org's UI, only with yellow highlights 😄

**Comment by timiil** [comment]
yes, that is sound very good. 
also, if we can set a pakcage 'max version count' value, automatic delete the oldest versions... is it OK ?

**Comment by bkraul** [comment]
How can we manually delete packages at this point? I attempted to delete with 
```
nuget delete MyPackage 1.0 -Source https://nuget.mydomain.com -ApiKey API-KEY
```

I get the warning and a prompt, and then I get this:
``` 
DELETE https://nuget.mydomain.com/api/v2/package/MyPackage/1.0
NoContent https://nuget.domain.com/api/v2/package/MyPackage/1.0 498ms
EPG.CommonLibrary 1.0.6.2 was deleted successfully.
```
But I can still see the package in the UI and the files still exist on the storage location.

**Comment by LordMike** [comment]
Deletion _should_ be implemented, at least the [code is there](https://github.com/loic-sharma/BaGet/blob/491f6bf178f265f0a1ab9d05df6e2e6adac55708/src/BaGet.Core/Services/PackageDeletionService.cs#L32-L42). The [default is however](https://github.com/loic-sharma/BaGet/blob/master/src/BaGet.Core/Configuration/BaGetOptions.cs#L27), for deletions to unlist packages - not delete them. 

You need to configure `BaGetOptions`, which if I [read this](https://github.com/loic-sharma/BaGet/blob/4f1591b061c8438ee100795c1206f02285225dd4/src/BaGet/Extensions/ServiceCollectionExtensions.cs#L36) correctly is the root of the `appsettings.json`.

Just adding `"PackageDeletionBehavior": "HardDelete"` in the `appsettings.json`, next to `ApiKey` should do it for you.

**Comment by bkraul** [comment]
I enabled the ```HardDelete``` option, and was able to delete the package properly.

However, what about package replacement? Should not there be an option to replace packages with matching version number? Right now, even with ```HardDelete``` set, I still get ```409``` when replacing the package. I read about the whole left-pad thing, and I understand it, but our organization's needs are different.

We are currently using phpNuget, which supports deletion and replacement, but we are seeing very slow activity in the development and are looking for other options.

Thanks for your help, @LordMike .

**Comment by LordMike** [comment]
This seems like a seperate issue - I can't find anything, but I think I once saw something along the lines of allowing replacement of packages. Because yes, BaGet is to be run in a local setting, and cater to approximately one use case.. (those that run it), so it should allow all kinds of weirdness.

I'd create an issue on it, @bkraul

**Comment by loic-sharma** [comment]
Thanks for the help @LordMike!

**Comment by Cocotus** [comment]
I also needed to "clean" up my packages and searched for a way to delete some old test packages from Baget. Unfortunately theres no information on the help page (like its done with PUSH) of my Baget installation, so Im glad I found this topic! :)

I used this command to delete a package from Baget:

**dotnet nuget delete mynuget 1.0.0 -s http://localhost:5000/v3/index.json -k "myapikey" --non-interactive**

Is it possible to add another help page in Baget Server for this? I think this one might be an option which is used by many people.

**Comment by jsreynolds** [comment]
I'd like to throw my hat in the ring as this being a really useful addition. I've been doing command-line jujitsu to remove packages, but for packages with multiple revisions this is a big PITA. Also, the removal removes the package but not its parent folder, which means I have to go behind and clean up the folder structure as well.

Loving the app btw - this is basically the only missing feature for me.

Best,

--J

**Comment by Darth-Jurassic** [comment]
> However, what about package replacement? Should not there be an option to replace packages with matching version number? Right now, even with `HardDelete` set, I still get `409` when replacing the package. I read about the whole left-pad thing, and I understand it, but our organization's needs are different. It would help if the UI allowed for proper package management, but I understand it is in early stages, and it is looking GREAT!

You may use "AllowPackageOverwrites": true, in the same appsettings.json, but this feature is extremely toxic. Consider applying proper versioning in your CI / CD process.

**Comment by Baklap4** [comment]
Any updates on this? We'd love to have a button to delete the package :)!

**Comment by Suriman** [comment]
Any update?

**Comment by a-a-k** [comment]
is this project still supported?

**Comment by OnSive** [comment]
// Push

I would love to have this feature

**Comment by nguyenloki258** [comment]
Any update?

**Comment by qwertz44** [comment]
push

**Comment by Joan-do** [comment]
Any update?

**Comment by lonix1** [comment]
Use gitea, which has a built-in nuget server.
